src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9843,12 +9843,76 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 else
                 {
                     expression = this.ParseSubExpression(Precedence.Expression);
+
+                    // See if the expression is an invocation that could also be successfully parsed and interpreted
+                    // as a deconstruction target. I.e. something like "var (x, y)" or "var (x, (y, z))".
+                    // We should report an error in this case because we plan to support deconstruction for out arguments
+                    // in the future. We need to ensure that we do not successfully interpret this as an invocation of a
+                    // ref-returning method named var with two, or more arguments. 
+                    if (IsDeconstructionCompatibleArgument(refOrOutKeyword, expression))
+                    {
+                        expression = this.AddError(expression, ErrorCode.ERR_OutVarDeconstructionIsNotSupported);
+                    }
                 }
             }
 
             return _syntaxFactory.Argument(nameColon, refOrOutKeyword, expression);
         }
 
+        private static bool IsDeconstructionCompatibleArgument(SyntaxToken refOrOutKeyword, ExpressionSyntax expression)
+        {
+            if (refOrOutKeyword?.Kind == SyntaxKind.OutKeyword &&
+                expression.Kind == SyntaxKind.InvocationExpression)
+            {
+                var invocation = (InvocationExpressionSyntax)expression;
+                ExpressionSyntax invocationTarget = invocation.Expression;
+
+                return invocationTarget.Kind == SyntaxKind.IdentifierName &&
```

Comment:
Reopening #13148 for LDM alignment.

---
In reply to: [77217479](https://github.com/dotnet/roslyn/pull/13467#discussion_r77217479) [](ancestors = 77217479)